src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int totalBuy = 0;
+        int totalSupply = 0;
+        int result = 0;
+        int value = 0;
+        String line;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            line = bufferedReader.readLine();
+            while (line != null) {
+                String[] parts = line.split(",");
+                value = Integer.parseInt(parts[1]);
+
+                if (parts[0].equals("buy")) {
+                    totalBuy += value;
+                } else if (parts[0].equals("supply")) {
+                    totalSupply += value;
+                }
+                line = bufferedReader.readLine();
+            }
+
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t read the file!");
```

Exceptions are rethrown without preserving the original cause. The checklist explicitly says: "Don't ignore exceptions ... Better re-throw `RuntimeException` with the original exception and some message in the parameters:" and shows the example `throw new RuntimeException("Can't read data from the file " + fileName, e);` 

Change this catch block to rethrow with the original exception as the cause, e.g. `throw new RuntimeException("Can't read data from file " + fromFileName, e);` so the root cause is preserved.